orbit/cmd/orbit/orbit.go

```diff
@@ -527,6 +534,61 @@ func main() {
 				log.Info().Err(err).Msg("Initial flags update failed")
 			}
 			g.Add(flagRunner.Execute, flagRunner.Interrupt)
+
+			if deviceClient.GetServerCapabilities().Has(fleet.CapabilityTokenRotation) {
+				log.Info().Msg("token rotation is enabled")
+
+				trw.SetRemoteUpdateFunc(func(token string) error {
+					return orbitClient.SetOrUpdateDeviceToken(orbitNodeKey, token)
+				})
+
+				if err := trw.LoadOrGenerate(); err != nil {
+					return fmt.Errorf("initializing token read writer: %w", err)
+				}
+
+				// perform an initial check to see if the token
+				// has not been revoked by the server
+				if err := deviceClient.CheckToken(trw.GetCached()); err != nil {
+					trw.Rotate()
+				}
+
+				go func() {
+					// This timer is used to check if the token should be rotated if  at
+					// least one hour has passed since the last modification of the token
+					// file.
+					//
+					// This is better than using a ticker that ticks every hour because the
+					// we can't ensure the tick actually runs every hour (eg: the computer is
+					// asleep).
+					rotationDuration := 30 * time.Second
+					rotationTicker := time.NewTicker(rotationDuration)
+					defer rotationTicker.Stop()
+
+					// This timer is used to periodically check if the token is valid. The
+					// server might deem a toked as invalid for reasons out of our control,
+					// for example if the database is restored to a back-up or if somebody
+					// manually invalidates the token in the db.
+					remoteCheckDuration := 5 * time.Minute
+					remoteCheckTicker := time.NewTicker(remoteCheckDuration)
+					defer remoteCheckTicker.Stop()
+
+					for {
+						select {
+						case <-rotationTicker.C:
+							if trw.HasExpired() {
+								log.Info().Msg("token TTL expired, rotating token")
+								trw.Rotate()
```

Comment:
We should log the returned `error` (in the `case` below too).

Reply:
great catch, thanks